src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,72 @@
-import React from 'react';
+import React, {
+  FormEvent, useCallback, useMemo, useState,
+} from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { ResponseError } from '../../types/ReponseError';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+interface Props {
+  addToList: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ addToList }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState<ResponseError | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubmit = useCallback((event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  }, []);
+
+  const handleSearchSubmit = useCallback(async () => {
+    try {
+      setIsLoading(true);
+
+      const movieData = await getMovie(query);
+
+      if ('Error' in movieData) {
+        setError(movieData);
+        setMovie(null);
+      } else {
+        const {
+          Title, Plot, imdbID, Poster,
+        } = movieData;
+
+        setMovie({
+          title: Title,
+          description: Plot,
+          imgUrl: Poster,
+          imdbId: imdbID,
+          imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+        });
+
+        setError(null);
+      }
```

Comment:
It's not the best practice to use if else structure, you definitely can simplify that to make the code more clear and readable.